tests/nested/manual/core20-auto-remove-user/task.yaml

```diff
@@ -0,0 +1,144 @@
+  sed -i "s/{VERSION}/$VERSION/g" ./user1-final.json
+
+  # sign the user1.json using gendeveloper1
+  gendeveloper1 sign-model < ./user1-final.json > user1.assert
+  
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  #shellcheck source=tests/lib/store.sh
+  . "$TESTSLIB"/store.sh
+
+  # Setup the fake-store for ubuntu-image to use when creating our core image.
+  # We immediately tear down the staging store, to make sure snapd is not pointed
+  # towards this once we invoke ubuntu-image. 
+  # We also need to point NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL towards the newly
+  # setup fake-store to actually have it use it.
+  setup_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+  teardown_staging_store
+
+  echo "Expose the needed assertions through the fakestore"
+  cp "$TESTSLIB"/assertions/developer1.account "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+  cp "$TESTSLIB"/assertions/developer1.account-key "$NESTED_FAKESTORE_BLOB_DIR/asserts"
+
+  KEY_NAME=$(tests.nested download snakeoil-key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  # Get the nested system version
+  VERSION="$(tests.nested show version)"
+
+  echo "Grab and prepare the gadget snap"
+  snap download --basename=pc --channel="$VERSION/edge" pc
+  unsquashfs -d pc-gadget pc.snap
+
+  echo "Sign the shim binary"
+  tests.nested secboot-sign gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+
+  echo "Add the prepare-device hook"
+  mkdir -p pc-gadget/meta/hooks
+  cp prepare-device pc-gadget/meta/hooks/prepare-device
+
+  # delay all refreshes for a week from now, as otherwise refreshes for our 
+  # snaps (which are asserted by the testrootorg authority-id) may happen, which
+  # will break things because the signing keys won't match, etc. and 
+  # specifically snap-bootstrap in the kernel snap from the store won't trust
+  # the seed keys to unlock the encrypted data partition in the initramfs
+  sed defaults.yaml -e "s/HOLD-TIME/$(date --date="next week" +%Y-%m-%dT%H:%M:%S%:z)/" >> \
+    pc-gadget/meta/gadget.yaml
+
+  snap pack pc-gadget/ "$(tests.nested get extra-snaps-path)"
+
+  #shellcheck disable=SC2148
+  systemd-run --collect --unit fakedevicesvc fakedevicesvc localhost:11029
+
+  tests.nested build-image core
+  tests.nested create-vm core
+
+restore: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+
+  systemctl stop fakedevicesvc
+
+  #shellcheck source=tests/lib/store.sh
+  . "$TESTSLIB"/store.sh
+  teardown_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+
+execute: |
+  if [ "$TRUST_TEST_KEYS" = "false" ]; then
+      echo "This test needs test keys to be trusted"
+      exit
+  fi
+
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  # wait until device is initialized and has a serial
+  tests.nested wait-for device-initialized
+
+  # ensure we are looking at the correct user
```

Comment:
This is probably a silly question but could we add a system-user assertion for a `user2` and observe from user1 that it gets added and removed? or will this not work because of the rules of adding users when there is already one etc?

Reply:
Good question, we could definitely do this but I did this to avoid expanding the current auto-import shell scripts, because currently it only takes one auto-import assertion and we use this for the primary user. 

I'll happily do this if you feel like that would be better